mesonbuild/coredata.py

```diff
@@ -301,12 +284,16 @@ def validate_value(self, value: T.Union[str, T.List[str]]) -> T.List[str]:
             mlog.deprecation(msg)
         for i in newvalue:
             if not isinstance(i, str):
-                raise MesonException(f'String array element "{newvalue!s}" is not a string.')
+                raise MesonException(f'String array element "{newvalue!s}" for option "{self.name}" is not a string.')
         if self.choices:
             bad = [x for x in newvalue if x not in self.choices]
             if bad:
-                raise MesonException('Options "{}" are not in allowed choices: "{}"'.format(
-                    ', '.join(bad), ', '.join(self.choices)))
+                if len(bad) == 1:
+                    raise MesonException('Value "{}" for option "{}" is not in allowed choices: "{}"'.format(
+                        bad[0], self.name, ', '.join(self.choices)))
+                else:
+                    raise MesonException('Values "{}" for option "{}" are not in allowed choices: "{}"'.format(
+                        ', '.join(bad), self.name, ', '.join(self.choices)))
```

Comment:
```suggestion
                raise MesonException('Value{} "{}" for option "{}" are not in allowed choices: "{}"'.format(
                's' if len(bad) > 1 else '', ', '.join(bad), self.name, ', '.join(self.choices)))
```

Less repeating, and since we're about to abort the performance difference in the `str.join` is irrelavent